ambari-server/src/main/java/org/apache/ambari/server/utils/SecretReference.java

```diff
@@ -150,6 +166,49 @@ public static void replacePasswordsWithReferences(Map<PropertyInfo.PropertyType,
     }
   }
 
+  /**
+   * Returns all password properties defined in the stacks specified by the given stack id's. Keys in the map are
+   * file names (e.g hadoop-env.xml) and values are property names.
+   * @param stackIds the stack ids to specify which stacks to look for
+   * @return A set multimap of password type properties.
+   * @throws IllegalArgumentException when a non-existing stack is specified
+   */
+  public static SetMultimap<String, String> getAllPasswordProperties(Collection<StackId> stackIds) throws IllegalArgumentException {
+    AmbariMetaInfo metaInfo = AmbariServer.getController().getAmbariMetaInfo();
+    Collection<StackInfo> stacks = stackIds.stream().map( stackId -> {
+      try {
+        return metaInfo.getStack(stackId);
+      }
+      catch (StackAccessException ex) {
+        throw new IllegalArgumentException(ex);
+      }
+    }).collect(toList());
```

Comment:
Can you please extract this lambda expression to a function that can be referenced and tested more easily?

Reply:
This is a very simple lambda (a simple method call within a try - catch - re-throw). I don't think it makes sense to decompose it further.